src/godot.cpp

```diff
@@ -526,6 +526,12 @@ void GDExtensionBinding::initialize_level(void *p_userdata, GDExtensionInitializ
 			doc_data.load_data();
 		}
 	}
+
+#ifdef DEV_ENABLED
+	if ((ModuleInitializationLevel)p_level == MODULE_INITIALIZATION_LEVEL_EDITOR) {
+		WARN_PRINT_ONCE("GDExtension loaded from a development build.");
+	}
```

Comment:
Thanks for adding this!

However, two things:

- I think we should do this at  `MODULE_INITIALIZE_LEVEL_SCENE`. If someone does accidentally ship a dev build, and they don't notice it in the editor, then perhaps their users will notice it when they run the game
- I think we should include the path to the library in the message, which you can get by doing this:
  ```
    String library_path;
    internal::gdextension_interface_get_library_path(internal::library, library_path._native_ptr());
  ```

Reply:
Awesome, that's exactly the kind of function I was looking for!

Somehow, it stopped printing a warning in the editor though - it _only_ prints successfully when running the game. I can't even get it to print on my old commit hash - i wonder if I misremember that this ever worked.

I guess it's not half bad since it's only relevant if you do actually run the game?

Reply:
Hm. Maybe it's too early for it to appear in the editor? Does it appear in the console when starting the editor, but just not make it into the editor itself?

Reply:
Oh yeah, the message does make it into the console.

Reply:
Maybe we should do this message at both `MODULE_INITIALIZATION_LEVEL_SCENE` and `MODULE_INITIALIZATION_LEVEL_EDITOR` if the 2nd one will get it into the editor? Although, it sounds like that doesn't happen reliably in your testing?

Maybe we could do `callable_mp_static(...).call_deferred()` to do the warning just a little bit later, such that the editor might catch it?

Reply:
The `callable_mp_static` version seems to work reliably! 
I think it's worth the slightly awkward workaround to see the warning in the actual editor. Let me know if my lambda based version works for you.

Reply:
> The `callable_mp_static` version seems to work reliably!

Great!

> I think it's worth the slightly awkward workaround to see the warning in the actual editor. Let me know if my lambda based version works for you.

I agree. IMO, a lambda is fine here, given the nature of this code.